modin/core/execution/ray/implementations/cudf_on_ray/partitioning/axis_partition.py

```diff
@@ -61,58 +51,31 @@ def reduce(self, func):
         -------
         cuDFOnRayDataframePartition
         """
-        keys = [partition.get_key() for partition in self.partitions]
-        gpu_managers = [partition.get_gpu_manager() for partition in self.partitions]
-        head_gpu_manager = gpu_managers[0]
-        cudf_dataframe_object_ids = [
-            gpu_manager.get.remote(key) for gpu_manager, key in zip(gpu_managers, keys)
-        ]
-
-        # FIXME: The signature of `head_gpu_manager.reduce` requires
-        # (first, others, func, axis=0, **kwargs) parameters, but `first`
-        # parameter isn't present.
-        key = head_gpu_manager.reduce.remote(
-            cudf_dataframe_object_ids, axis=self.axis, func=func
-        )
-        key = ray.get(key)
-        result = cuDFOnRayDataframePartition(gpu_manager=head_gpu_manager, key=key)
+        if self.axis == 0:
```

Comment:
Why do we have so many changes here?

Reply:
For some reason `reduce` has different implementations for columns and rows. I moved these implementations to the base class using `axis` flag.

If you look at the number of `FIXME` labels, then it is quite possible to rewrite this place so that there are no differences in implementation. I do not undertake to rewrite myself, I am not familiar with this code.

Reply:
It seems `RowPartition` and `ColumnPartition` classes do introduce specific logic for their defined axes after all. That's a point to think of the need to remove/keep these classes.